02-oop-inheritance/js/script.js

```diff
@@ -0,0 +1,82 @@
+var ObserverList = function () {
+    this.observerList = [];
+};
+
+ObserverList.prototype = {
+
+    constructor: ObserverList,
+
+    subscribeObserver: function (observer) {
+        this.observerList.push(observer);
+    },
+
+    unsubscribeObserver: function (observer) {
+        this.observerList.splice(this.observerList.indexOf(observer), 1);
+    }
+};
+
+var Movie = function (title,year,observer) {
+    this.title = title;
+    this.year = year;
+    ObserverList.call(observer);
+};
+
```

Comment:
You forgot to set ``Movie`` prototype as ``ObserverList``'s prototype (``Movie.prototype = Object.create(ObserverList.prototype)``). Otherwise, you'll not have ``ObserverList`` class methods available in your ``Movie``class.